eng/pipelines/templates/jobs/smoke.tests.yml

```diff
@@ -185,7 +185,22 @@ jobs:
           env:
             REQUIREMENTS: $(requirements)
 
-        - pwsh: pip install -r "$env:REQUIREMENTS" --no-deps --upgrade
+        - pwsh: |
+            $timeout = New-TimeSpan -Seconds 600
+            $stopwatch = [System.Diagnostics.Stopwatch]::StartNew()
+            do {
+              $failure = $false
+              try {
+                $ErrorActionPreference = "Stop"
+                pip --retries 10 install -r "$env:REQUIREMENTS" --no-deps --upgrade --no-cache-dir 2>&1
+              } catch {
+                $failure = $true
+                Write-Host "Installation failed, retrying in 3 minutes..."
+                sleep 180
+              } finally {
+                $ErrorActionPreference = "Continue"
+              }
+            } while (($stopwatch.elapsed -lt $timeout) -and ($failure))
```

Comment:
Probably not necessary to have a stopwatch and a timeout value but instead just keep the sleep on the failure case to help control the time and just have a retry count.